libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -1258,7 +1272,20 @@ void IotMqtt_Disconnect( IotMqttConnection_t mqttConnection,
 
     if( initCalled == true )
     {
-        /* Close the underlying network connection. This also cleans up keep-alive. */
+        /* Coverity finds a USE_AFTER_FREE error at this line. This is a false positive.
+         *
+         * This error is triggered by a dereference of 'mqttConnection' in
+         * '_IotMqtt_CloseNetworkConnection'. Coverity assumes that 'mqttConnection'
+         * was freed in '_IotMqtt_CreateOperation' above, where cleanup code will
+         * free 'pNewMqttConnection' upon allocation failure.
+         *
+         * This will never happen as 'mqttConnection' is allocated with a reference
```

Comment:
Minor correction here: `This will never happen as a valid MQTT connection passed to this function always has a positive reference count; therefore, '_IotMqtt_CreateOperation' will not free it. Only unreferenced MQTT connections will be freed.`

Reply:
ok. Sounds good